docs/boogie-code.md

```diff
@@ -38,6 +38,28 @@ void assume(bool v) {
 }
 ````
 SMACK interprets the `@` symbol by inserting the value of the C variable `v`.
+Since the arguments of variadic functions may be promoted, SMACK allows users
+to append a format character to the `@` symbol as an indicator that the argument,
+rather than its promoted form, should be used in the format string. There are four
+format characters: `'c'`, `'h'`, `'i'`, and `'f'`, which represent that the
+original types should be `i8`, `i16`, `i32`, and `float`. For example, without using
+such annotations, the definition of the `floorf` function is:
+```C
+float floorf(float x) {
+  double ret = __VERIFIER_nondet_double();
+  __SMACK_code("@ := ftd($rmode, $round.bvfloat(RTN, dtf($rmode, @)));", ret, x);
+   return ret;
+ }
+```
+, where the `ftd` and `dtf` functions are conversions between `float` and `double` variables.
```

Comment:
Remove this comma in the beginning of this line.

Reply:
Also "where the `ftd` and `dtf` functions are conversions between `float` and `double` variables that are needed to deal with the promotion of the arguments."